### ISSUE TITLE
The support for PS/2 mice and trackpads is broken in OpenDuet

### ISSUE THREAD

**Issue by vit9696** [acidanthera/bugtracker#1236]
Happens with QEMU and several other platforms. CC @Goldfish64

**Comment by Goldfish64** [comment]
Leaving open until PS/2 trackpad issues are resolved.

**Comment by nms42** [comment]
What is wrong with stuff from audk/MdeModulePkg/Bus/Isa/IsaBusDxe? Right now I have two (out of three) legacy notebooks with dead keyboard. Git logs did not reveal the reason for SioBusDxe usage.

**Comment by vit9696** [comment]
It requires `gEfiIsaHcProtocolGuid` protocol implemented by the firmware. However there is no public (and to be honest, I do not know of private either) implementation of this protocol. For this reason IsaBusDxe is basically unusable.

**Comment by nms42** [comment]
For the record, current version (0.6.3) works with Ps2KeyboardDxe.efi from 0.6.2. Did not checked mouse.

**Comment by vit9696** [comment]
Hrm. Do you mean that 0.6.3 does not work with Ps2KeyboardDxe from 0.6.3 but does with the one from 0.6.2 or what? Basically we fixed some PS/2 issues with the mice in 0.6.3, but not all, thus the issue is still open. However, I do not believe there is any regression in 0.6.3 (or any previous releases) in that regard.

**Comment by nms42** [comment]
I have three legacy notebooks - (1) Toshiba Satellite L830, (2) (eMachines|Acer) G730 and (3) Acer Aspire One AOD255.

OK, here is list of actions done so far:

On comp #2 version 0.6.3 boots OC & macos 10.13.6. So far, so good.

Comp #1 did not boot even OC itself. I made few restarts from scratch (mbr+fat32 flash; install boot0, ... by .command script). Will investigate more.

Comp #3 boot OC 0.6.3 but it became dead on selection menu. Replacing Ps2KeyboardDxe.efi with the one from 0.6.2 unlocked menu.

The situation is unclear and confusing yet.

**Comment by vit9696** [comment]
Right, for the 3rd one if there is no mistake in Ps2KeyboardDxe from 0.6.3 being non-functional, could you try to find the exact commit it started to fail. See the per-commit binaries here: https://dortania.github.io/builds/?product=OpenCorePkg&viewall=true

**Comment by nms42** [comment]
I did not checked legacy boot on Toshiba and AOD255 before at all.

Do DEBUG|NOOPT build of OpenDuet produce visible trace of activity?

**Comment by vit9696** [comment]
Not at the moment. Basically our general approach to debug OpenDuet is to insert SerialPortWrite calls in the relevant places. @Goldfish64 has a rather large park of machines, and in general Intel ones work more or less fine at least.

**Comment by nms42** [comment]
Well, I would like to recall all that rant about boot problems. Sorry for noise. Looks like it was a problem of boot media. I used SD card with usb card reader. After switching to noname usb flash with proper performed procedures all boot checks went well. WARNING! I checked keyboard only.
